pkgs/applications/networking/soulseek/nicotine-plus/default.nix

```diff
@@ -0,0 +1,57 @@
+{ stdenv, fetchFromGitHub, python27Packages, lib
+, geoip ? null
+, geolite-legacy ? null
+
+# Nicotine+ supports a country code blocker. This requires a (GPL'ed) library called GeoIP.
+, enableGeoIP ? true
+}:
+
+assert enableGeoIP -> python27Packages.GeoIP != null;
+assert enableGeoIP -> geoip                  != null;
+assert enableGeoIP -> geolite-legacy         != null;
+
+with stdenv.lib;
+
+python27Packages.buildPythonApplication rec {
+  pname = "nicotine-plus";
+  version = "1.4.1";
+  src = fetchFromGitHub {
+    owner = "Nicotine-Plus";
+    repo = "nicotine-plus";
+    rev = "4e057d64184885c63488d4213ade3233bd33e67b";
+    sha256 = "11j2qm67sszfqq730czsr2zmpgkghsb50556ax1vlpm7rw3gm33c";
+  };
+
+  propagatedBuildInputs = with python27Packages; [
+    pygtk
+    miniupnpc
+    mutagen
+    notify
+  ] ++ lib.lists.optional enableGeoIP
+     (GeoIP.overrideAttrs (_: {
+       propagatedBuildInputs = lib.lists.singleton
```

Comment:
This is definitely not the way to go. Should override the `geoip` argument instead.

Reply:
Sorry for delay.

You mean, something like this:

```
  ] ++ lib.lists.optional enableGeoIP
     (GeoIP.overrideAttrs (_: {
       propagatedBuildInputs = [geoip];
     }));
```

And:
```
  nicotine-plus = callPackage ../applications/networking/soulseek/nicotine-plus {
    geoip = geoipWithDatabase;
  };
```
?

Otherwise I don't know how to build `python27Packages.GeoIP` with `geoip` with `geolite-legacy`.

Reply:
Sorry, I did not explain enough. I meant something like `GeoIP.override { inherit geoip; }`.

Reply:
Thanks.